pkg/transform/kubernetes/kubernetes.go

```diff
@@ -151,6 +152,16 @@ func (t *Transformer) TransformServices(services []object.Service) ([]runtime.Ob
 			return nil, fmt.Errorf("failed to create deployments: %s", err)
 		}
 		result = append(result, objects...)
+
+		// apply labels to all objects created by the service
+		for _, o := range result {
+			switch t := o.(type) {
+			case *api_v1.Service:
+				transformerUtil.ApplyOpenComposeLabelsToProvider(&service.Labels, &t.ObjectMeta.Labels)
+			case *ext_v1beta1.Deployment:
+				transformerUtil.ApplyOpenComposeLabelsToProvider(&service.Labels, &t.ObjectMeta.Labels)
```

Comment:
I don't know a better way of doing this, the only problem with this method is that it causes problems like we had in kompose. If new k8s object generation support is added and if label application is not done here for that object we now have inconsistent objects generated, which is again hard to test.

If there is an alternate generic method to do this I would love to see that, which wil save us from mistakes in future.